### PR TITLE
MGDCTRS-348 Reduce the exposed attributes for Debezium connectors in the catalog

### PR DESCRIPTION
**Pull request by ani-sha**
https://issues.redhat.com/browse/MGDCTRS-348

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,5 @@ release.properties
 
 generated-sources/
 *.log
+
+.idea/
```
Comment (rk3rn3r) [inline .gitignore:13]: This directory is environment specific, please move it to your global `.gitignore` in your user's home dir.

File: src/main/java/io/debezium/mcs/envelope/ConnectorTypesEnvelope.java [hunk 2]
```diff
@@ -25,7 +25,7 @@
     private static final String BASE_IMAGES_HREF = "http://example.com/images/";
 
     private static final Schema PASSWORD_SCHEMA_PLAINTEXT = new SchemaImpl();
-    private static final Schema PASSWORD_SCHEMA_REFRENCE = new SchemaImpl();
+    private static final Schema PASSWORD_SCHEMA_REFERENCE = new SchemaImpl();
```
Comment (rk3rn3r) [inline src/main/java/io/debezium/mcs/envelope/ConnectorTypesEnvelope.java:28]: good catch :see_no_evil:

File: descriptors/mongodb/debezium-mongodb-1.9.0.Alpha1.json [hunk 3]
```diff
@@ -62,93 +54,6 @@
...
-        "connect.backoff.max.delay.ms" : {
-          "format" : "int64",
-          "title" : "Maximum delay before reconnection (ms)",
-          "description" : "The maximum delay when trying to reconnect to a primary after a connection cannot be made or when no primary is available, given in milliseconds. Defaults to 120 second (120,000 ms).",
-          "default" : 120000,
-          "type" : "integer",
-          "x-name" : "connect.backoff.max.delay.ms",
-          "x-category" : "CONNECTION_ADVANCED"
-        },
-        "mongodb.authsource" : {
```
Comment (rk3rn3r) [inline descriptors/mongodb/debezium-mongodb-1.9.0.Alpha1.json:117]: should be kept

File: descriptors/mysql/debezium-mysql-1.9.0.Alpha1.json [hunk 4]
```diff
@@ -27,7 +27,7 @@
           "format" : "int64",
           "title" : "Cluster ID",
           "description" : "A numeric ID of this database client, which must be unique across all currently-running database processes in the cluster. This connector joins the MySQL database cluster as another server (with this unique ID) so it can read the binlog. By default, a random number is generated between 5400 and 6400.",
-          "default" : 5685,
```
Comment (rk3rn3r) [inline descriptors/mysql/debezium-mysql-1.9.0.Alpha1.json:30]: docs create that automatically. you can undo this change.

File: descriptors/mysql/debezium-mysql-1.9.0.Alpha1.json [hunk 5]
```diff
@@ -259,336 +131,13 @@
...
-        },
-        "tombstones.on.delete" : {
-          "title" : "Change the behaviour of Debezium with regards to delete operations",
-          "description" : "Whether delete operations should be represented by a delete event and a subsquenttombstone event (true) or only by a delete event (false). Emitting the tombstone event (the default behavior) allows Kafka to completely delete all events pertaining to the given key once the source record got deleted.",
-          "default" : true,
-          "type" : "boolean",
-          "x-name" : "tombstones.on.delete",
-          "x-category" : "CONNECTOR"
-        },
-        "decimal.handling.mode" : {
```
Comment (rk3rn3r) [inline descriptors/mysql/debezium-mysql-1.9.0.Alpha1.json:346]: @gunnarmorling as this is not in the meeting notes, I think it should be included?

File: descriptors/postgres/debezium-postgres-1.9.0.Alpha1.json [hunk 6]
```diff
@@ -72,83 +72,6 @@
...
-          "x-category" : "CONNECTION_ADVANCED"
-        },
-        "database.initial.statements" : {
-          "title" : "Initial statements",
-          "description" : "A semicolon separated list of SQL statements to be executed when a JDBC connection to the database is established. Note that the connector may establish JDBC connections at its own discretion, so this should typically be used for configurationof session parameters only, but not for executing DML statements. Use doubled semicolon (';;') to use a semicolon as a character and not as a delimiter.",
-          "type" : "string",
-          "x-name" : "database.initial.statements",
-          "x-category" : "CONNECTION_ADVANCED"
-        },
-        "plugin.name" : {
```
Comment (rk3rn3r) [inline descriptors/postgres/debezium-postgres-1.9.0.Alpha1.json:143]: @gunnarmorling I think `plugin.name` should be included?

File: descriptors/postgres/debezium-postgres-1.9.0.Alpha1.json [hunk 7]
```diff
@@ -157,57 +80,6 @@
           "x-name" : "slot.name",
           "x-category" : "CONNECTION_ADVANCED_REPLICATION"
         },
-        "slot.stream.params" : {
```
Comment (rk3rn3r) [inline descriptors/postgres/debezium-postgres-1.9.0.Alpha1.json:160]: @gunnarmorling I am not sure how often `slot.stream.params` are used. include or exclude?

File: descriptors/postgres/debezium-postgres-1.9.0.Alpha1.json [hunk 8]
```diff
@@ -290,284 +154,13 @@
...
-        },
-        "tombstones.on.delete" : {
-          "title" : "Change the behaviour of Debezium with regards to delete operations",
-          "description" : "Whether delete operations should be represented by a delete event and a subsquenttombstone event (true) or only by a delete event (false). Emitting the tombstone event (the default behavior) allows Kafka to completely delete all events pertaining to the given key once the source record got deleted.",
-          "default" : true,
-          "type" : "boolean",
-          "x-name" : "tombstones.on.delete",
-          "x-category" : "CONNECTOR"
-        },
-        "decimal.handling.mode" : {
```
Comment (rk3rn3r) [inline descriptors/postgres/debezium-postgres-1.9.0.Alpha1.json:358]: Should be included according to the meeting notes? (see comment above)

File: descriptors/postgres/debezium-postgres-1.9.0.Alpha1.json [hunk 9]
```diff
@@ -290,284 +154,13 @@
...
-        },
-        "include.schema.comments" : {
-          "title" : "Include Table and Column Comments",
-          "description" : "Whether the connector parse table and column's comment to metadata object.Note: Enable this option will bring the implications on memory usage. The number and size of ColumnImpl objects is what largely impacts how much memory is consumed by the Debezium connectors, and adding a String to each of them can potentially be quite heavy. The default is 'false'.",
-          "default" : false,
-          "type" : "boolean",
-          "x-name" : "include.schema.comments",
-          "x-category" : "CONNECTOR"
-        },
-        "interval.handling.mode" : {
```
Comment (rk3rn3r) [inline descriptors/postgres/debezium-postgres-1.9.0.Alpha1.json:393]: @gunnarmorling I would leave `interval.handling.mode` as excluded, but should the Debezium Operator maybe set the value to "string"?

File: descriptors/sqlserver/debezium-sqlserver-1.9.0.Alpha1.json [hunk 10]
```diff
@@ -71,35 +71,6 @@
           "x-name" : "database.dbname",
           "x-category" : "CONNECTION"
         },
-        "database.names" : {
```
Comment (rk3rn3r) [inline descriptors/sqlserver/debezium-sqlserver-1.9.0.Alpha1.json:74]: @ani-sha I think this must be included.

File: descriptors/sqlserver/debezium-sqlserver-1.9.0.Alpha1.json [hunk 11]
```diff
@@ -71,35 +71,6 @@
...
-          "format" : "list,regex",
-          "title" : "Databases",
-          "description" : "The names of the databases from which the connector should capture changes",
-          "type" : "string",
-          "x-name" : "database.names",
-          "x-category" : "CONNECTION"
-        },
-        "database.instance" : {
```
Comment (rk3rn3r) [inline descriptors/sqlserver/debezium-sqlserver-1.9.0.Alpha1.json:82]: same here?

File: descriptors/sqlserver/debezium-sqlserver-1.9.0.Alpha1.json [hunk 12]
```diff
@@ -149,260 +112,13 @@
           "x-name" : "snapshot.mode",
           "x-category" : "CONNECTOR_SNAPSHOT"
         },
-        "snapshot.isolation.mode" : {
```
Comment (rk3rn3r) [inline descriptors/sqlserver/debezium-sqlserver-1.9.0.Alpha1.json:152]: @gunnarmorling I never used SQL Server. How important is that property for SQL Server? include or exclude it?

File: descriptors/sqlserver/debezium-sqlserver-1.9.0.Alpha1.json [hunk 13]
```diff
@@ -149,260 +112,13 @@
...
-        },
-        "tombstones.on.delete" : {
-          "title" : "Change the behaviour of Debezium with regards to delete operations",
-          "description" : "Whether delete operations should be represented by a delete event and a subsquenttombstone event (true) or only by a delete event (false). Emitting the tombstone event (the default behavior) allows Kafka to completely delete all events pertaining to the given key once the source record got deleted.",
-          "default" : true,
-          "type" : "boolean",
-          "x-name" : "tombstones.on.delete",
-          "x-category" : "CONNECTOR"
-        },
-        "decimal.handling.mode" : {
```
Comment (rk3rn3r) [inline descriptors/sqlserver/debezium-sqlserver-1.9.0.Alpha1.json:227]: According to the meeting notes this should be included.

File: descriptors/sqlserver/debezium-sqlserver-1.9.0.Alpha1.json [hunk 14]
```diff
@@ -429,64 +145,6 @@
...
-        },
-        "transaction.topic" : {
-          "title" : "Transaction topic name",
-          "description" : "The name of the transaction metadata topic. The placeholder ${database.server.name} can be used for referring to the connector's logical name; defaults to ${database.server.name}.transaction.",
-          "default" : "${database.server.name}.transaction",
-          "type" : "string",
-          "x-name" : "transaction.topic",
-          "x-category" : "ADVANCED"
-        },
-        "database.server.timezone" : {
```
Comment (rk3rn3r) [inline descriptors/sqlserver/debezium-sqlserver-1.9.0.Alpha1.json:484]: @gunnarmorling same question here, in the Windows world, this might be an important topic, especially as even system clocks are set to local timezones often, is it? include or exclude?

**Comment by rk3rn3r** [comment]
@ani-sha looks good except for some properties. I added comments where we need to change based on the meeting notes and I asked Gunnar about some properties where I was not sure if we should include or exclude them.

File: .gitignore (hunk 1 above)
Comment (gunnarmorling) [reply .gitignore:13]: Nah, it's fine here. Let's make life simplers for all participating devs, not harder.

File: descriptors/sqlserver/debezium-sqlserver-1.9.0.Alpha1.json (hunk 11 above)
Comment (ani-sha) [reply descriptors/sqlserver/debezium-sqlserver-1.9.0.Alpha1.json:82]: Gunnar suggested not including it.

File: .gitignore (hunk 1 above)
Comment (rk3rn3r) [reply .gitignore:13]: It's not harder @gunnarmorling, Sorry, I disagree! That's just how you usually do it, based on your environment you put your IDE specific entries into your user's `.gitignore` file in the home dir. We should not put all temp directories for all possible IDEs into the project's `.gitignore`. DRY!

File: descriptors/postgres/debezium-postgres-1.9.0.Alpha1.json (hunk 6 above)
Comment (rk3rn3r) [reply descriptors/postgres/debezium-postgres-1.9.0.Alpha1.json:143]: @gunnarmorling what is with that one?

File: descriptors/sqlserver/debezium-sqlserver-1.9.0.Alpha1.json (hunk 10 above)
Comment (rk3rn3r) [reply descriptors/sqlserver/debezium-sqlserver-1.9.0.Alpha1.json:74]: @ani-sha do you have notes about the outcome? Added you the decisions to the meeting notes?

**Comment by rk3rn3r** [comment]
LGTM @ani-sha, thx a lot! I also added your spreadsheet to the meeting notes document.
Could you please rebase your branch and re-build the descriptors? I had to change something around the `data_shape` property. Then we are good to merge the PR! Feel free to do squash and merge yourself then.

File: descriptors/postgres/debezium-postgres-1.9.0.Alpha1.json (hunk 6 above)
Comment (gunnarmorling) [reply descriptors/postgres/debezium-postgres-1.9.0.Alpha1.json:143]: No, let's just go with pgoutput as the only supported one. wal2json is unsupported soon anyways, and decoderbufs isn't needed here either.

File: descriptors/postgres/debezium-postgres-1.9.0.Alpha1.json (hunk 9 above)
Comment (gunnarmorling) [reply descriptors/postgres/debezium-postgres-1.9.0.Alpha1.json:393]: Ok, makes sense. We should then set the default to string for Debezium 2.0 in general though. Can you add this to the DBZ 2.0 issue?

File: descriptors/sqlserver/debezium-sqlserver-1.9.0.Alpha1.json (hunk 10 above)
Comment (gunnarmorling) [reply descriptors/sqlserver/debezium-sqlserver-1.9.0.Alpha1.json:74]: database.names is part of the currently being worked-on multiple DB support for this connector. It's not ready for prime-time yet.

File: descriptors/sqlserver/debezium-sqlserver-1.9.0.Alpha1.json (hunk 11 above)
Comment (gunnarmorling) [reply descriptors/sqlserver/debezium-sqlserver-1.9.0.Alpha1.json:82]: Yeah, let's skip it for now. We can add it later on if needed.

File: descriptors/sqlserver/debezium-sqlserver-1.9.0.Alpha1.json (hunk 12 above)
Comment (gunnarmorling) [reply descriptors/sqlserver/debezium-sqlserver-1.9.0.Alpha1.json:152]: Let's leave it out for now.

File: descriptors/sqlserver/debezium-sqlserver-1.9.0.Alpha1.json (hunk 14 above)
Comment (gunnarmorling) [reply descriptors/sqlserver/debezium-sqlserver-1.9.0.Alpha1.json:484]: I don't think we need it for SP. We can easily add those things later on.
